server/src/main/java/io/crate/planner/operators/JoinConditionSymbolsExtractor.java

```diff
@@ -98,7 +97,7 @@ public Void visitFunction(Function function, Context context) {
                     context.insideEqOperator = true;
                     int duplicatePos = 0;
                     for (Symbol arg : function.arguments()) {
-                        var relation = arg.accept(RELATION_EXTRACTOR, null);
+                        var relation = getOnlyElement(RelationNameCollector.collect(arg));
```

Comment:
Is it save to use `getOnlyElement` here as compared to the removed visitor it will throw an exception if more than 1 relation is collected?

Reply:
The exact behaviour of the original visitor would be to get the first element or `null`. However, haven't found any case so far where the current code breaks.

Reply:
I checked also quickly, and I think we cannot get there, as then we don't fulfill the equijoin detection so it will exit in the beginning of `visitFunction`.